src/Microsoft.TestPlatform.ObjectModel/DataCollector/IConfigurableDataCollectorAttachmentProcessor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Xml;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+    /// <summary>
+    /// Interface for data collectors add-ins that choose to reprocess generated attachments
+    /// </summary>
+    public interface IConfigurableDataCollectorAttachmentProcessor
```

Comment:
>IConfigurableDataCollectorAttachmentProcessor [](http://example.com/codeflow?start=21&length=45)

So the one difference between this and the existing `IDataCollectorAttachmentProcessor` interface is that we now pass in the data collector configuration in the ProcessAttachmentSetsAsync method? Do we know if this interface is (or even can be) used by anyone outside? I wonder if we can just change the existing interface if this wasn't usable externally anyway. Also would we need the whole runsettings blob instead of just that data collectors config. Would someone be interested in reading anything in the runconfiguration element for instance?

Reply:
>  wonder if we can just change the existing interface if this wasn't usable externally anyway. 

We cannot update the existing one(we can do addition of new method but looks weird because for the implementer only one out of the two method will be called so it's not clear imo from user perspective) because if we implement new one but we load collector extension inside old SDK hosts we'll throw a missing implementation exception.

> Also would we need the whole runsettings blob instead of just that data collectors config. Would someone be interested in reading anything in the runconfiguration element for instance?

We had this discussion with @jakubch1 and @fhnaseer and we can do it, but again it could look strange from user perspective, inside DataCollector we're flowing only collector section config and not all runsettings so it's a bit asymmetric. And we expect that settings for an attachment processor that is "attached" to specific collector should handle that "context" and so settings should come from datacollector section.

Anyway we're open to discuss this one if you strongly feel the need.

Reply:
Let's break existing interface as nobody is using it now and regarding runsettings I think we should be consistent and pass only part for this collector. If some additional options are needed we will need to add options to this data collector.

Reply:
yeah, adding additional runsettings data would need to break the interface again, so I was hoping we could just do that once.